src/evict/evict_page.c

```diff
@@ -326,6 +319,9 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
 err:
         if (!closing)
             __evict_exclusive_clear(session, ref, previous_state);
+
+        if (ebusy_only && ret != EBUSY)
```

Comment:
Should we have this check before the call to `__evict_exclusive_clear`? Do we want to clear if things are already going wrong?

Reply:
I prefer to always clean up the resources. I don't think whether we do clear or not matters but I prefer it to be cleared.

Reply:
If we don't clean this up, the page will remain locked. I agree that it probably does not matter whether we unlock it at this point, but something in me cringes at the idea of not releasing locks...